src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -6,77 +6,291 @@ description: Learn how to add a single Flutter screen to your existing iOS app.
 
+**In `AppDelegate.swift`:**
+
+<?code-excerpt "AppDelegate.swift" title?>
 ```swift
 import UIKit
 import Flutter
-import FlutterPluginRegistrant // Only if you have Flutter Plugins.
+import FlutterPluginRegistrant // Used to connect plugins.
 
 @UIApplicationMain
-class AppDelegate: FlutterAppDelegate {
-  var flutterEngine : FlutterEngine?;
-  // Only if you have Flutter plugins.
+class AppDelegate: FlutterAppDelegate { // More on the FlutterAppDelegate below.
+  lazy var flutterEngine = FlutterEngine(name: "my flutter engine")
+
   override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-    self.flutterEngine = FlutterEngine(name: "io.flutter", project: nil);
-    self.flutterEngine?.run(withEntrypoint: nil);
+    // Runs the default Dart entrypoint with a default Flutter route.
+    flutterEngine.run();
     GeneratedPluginRegistrant.register(with: self.flutterEngine);
     return super.application(application, didFinishLaunchingWithOptions: launchOptions);
   }
+}
+```
+
+</div>
+
+</div>{% comment %} FlutterEngine language tab end {% endcomment -%}
 
+### Show a FlutterViewController with your FlutterEngine
+
+The following example shows a generic ViewController with a UIButton hooked to
+present a [`FlutterViewController`]({{site.api}}/objcdoc/Classes/FlutterViewController.html).
+The `FlutterViewController` uses the `FlutterEngine` instance created in the
+`AppDelegate`.
+
+<ul class="nav nav-tabs" id="vc-language" role="tablist">
+  <li class="nav-item">
+    <a class="nav-link active" id="vc-objc" href="#vc-objc-tab" role="tab" aria-controls="vc-objc" aria-selected="true">Objective-C</a>
+  </li>
+  <li class="nav-item">
+    <a class="nav-link" id="vc-swift" href="#vc-swift-tab" role="tab" aria-controls="vc-swift" aria-selected="false">Swift</a>
+  </li>
+</ul>
+
+<div class="tab-content"> {% comment %} FlutterViewController language tab start {% endcomment -%}
+
+<div class="tab-pane active" id="vc-objc-tab" role="tabpanel" aria-labelledby="vc-objc-tab" markdown="1">
+
+<?code-excerpt "ViewController.m" title?>
+```objectivec
+@import Flutter;
+#import "AppDelegate.h"
+#import "ViewController.h"
+
+@implementation ViewController
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    // Make a button to call the showFlutter function below when pressed.
+    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
+    [button addTarget:self
+               action:@selector(showFlutter)
+     forControlEvents:UIControlEventTouchUpInside];
+    [button setTitle:@"Show Flutter!" forState:UIControlStateNormal];
+    [button setBackgroundColor:[UIColor blueColor]];
+    button.frame = CGRectMake(80.0, 210.0, 160.0, 40.0);
+    [self.view addSubview:button];
+}
+
+- (void)showFlutter {
+    FlutterEngine *flutterEngine =
+        [(AppDelegate *)[[UIApplication sharedApplication] delegate] flutterEngine];
```

Comment:
```objective-c
UIApplication.sharedApplication.delegate.flutterEngine
```

Reply:
<I didn't try it yet> don't we need to keep the cast to AppDelegate? </>

Reply:
Oh yeah `((AppDelegate *)UIApplication.sharedApplication.delegate).flutterEngine`

Reply:
done